test/writer.test.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+var Expect = require('chai').expect;
+
+var Writer = require('../lib/writer');
+
+var VERSION = '1.2.3';
+
+describe('writer', function () {
+
+  describe('markdown', function () {
+
+    it('makes heading h2 if major version', function () {
+      var options = { major: true };
+
+      return Writer.markdown(VERSION, [], options)
+      .then(function (changelog) {
+        var heading = changelog.split('\n')[0];
+
+        Expect(heading).to.contain('## ' + VERSION);
+      });
+    });
+
+    it('makes heading h3 if minor version', function () {
+      var options = { minor: true };
+
+      return Writer.markdown(VERSION, [], options)
+      .then(function (changelog) {
+        var heading = changelog.split('\n')[0];
+
+        Expect(heading).to.contain('### ' + VERSION);
+      });
+    });
+
+    it('makes heading h4 if minor version', function () {
+      var options = { patch: true };
+
+      return Writer.markdown(VERSION, [], options)
+      .then(function (changelog) {
+        var heading = changelog.split('\n')[0];
+
+        Expect(heading).to.contain('#### ' + VERSION);
+      });
+    });
+
+    it('flushes out a commit type with its full name', function () {
+      var commits = [
+        { type: 'feat', category: 'testing', subject: 'did some testing', hash: '1234567890' }
+      ];
+
+      return Writer.markdown(VERSION, commits, {})
+      .then(function (changelog) {
+        return changelog.split('\n');
+      })
+      .filter(function (line) {
+        return line.indexOf('#####') > -1;
+      })
+      .get(0)
+      .then(function (line) {
+        Expect(line).to.eql('##### New Features');
+      });
+    });
+
+    it('uses the default type name for uncommon types', function () {
+      var commits = [
+        { type: 'uncommon', category: 'testing', subject: 'did some testing', hash: '1234567890' }
+      ];
+
+      return Writer.markdown(VERSION, commits, {})
+      .then(function (changelog) {
+        return changelog.split('\n');
+      })
+      .filter(function (line) {
+        return line.indexOf('#####') > -1;
+      })
+      .get(0)
+      .then(function (line) {
+        Expect(line).to.eql('##### Other Changes');
+      });
+    });
+
+    it('groups all uncommon types together', function () {
+      var commits = [
+        { type: 'uncommon', category: 'testing', subject: 'did some testing', hash: '1234567890' },
+        { type: 'unknown', category: 'testing', subject: 'did some more testing', hash: '1234567890' }
+      ];
+
+      return Writer.markdown(VERSION, commits, {})
+      .then(function (changelog) {
+        return changelog.split('\n');
+      })
+      .filter(function (line) {
+        return line.indexOf('#####') > -1;
+      })
+      .tap(function (lines) {
+        Expect(lines).to.have.length(1);
```

Comment:
This isn't looking for commits, it's looking for headings e.g. "New Features", "Bug Fixes", etc. This test is making sure that all `others` is grouped under the same heading of "Other Changes" as opposed to creating two headings, both of which say "Other Changes". Does that make sense?